src/plugins/data_source_management/public/components/data_source_menu/types.ts

```diff
@@ -40,6 +41,7 @@ export interface DataSourceMenuProps<T = any> {
   uiSettings?: IUiSettingsClient;
   application?: ApplicationStart;
   setMenuMountPoint?: (menuMount: MountPoint | undefined) => void;
+  dataSourceSelection: DataSourceSelection;
```

Comment:
I think this should be part of componentConfig since not all components need to provide this as prop. Also, enforcing this prop as required would break all consumers, since it is provided by the DSM plugin, we can make it optional?

Reply:
I prefer to let it out of componentConfig because this instance should be passed from DSM not other plugins. As this instance is shared and limited in DSM, so we can set it as strict required.

I initially thought that other plugins only can render DS selector components through the two functions `ui.DataSourceSelector` and `ui.getDataSourceMenu` (like this way https://github.com/opensearch-project/OpenSearch-Dashboards/blob/main/src/plugins/dev_tools/public/application.tsx#L118), so I only need to pass the dataSourceSelection instance to the component through this two functions of this DSM.

Until I saw home plugin directly importing component class for rendering without using two functions above(https://github.com/opensearch-project/OpenSearch-Dashboards/blob/main/src/plugins/home/public/application/components/tutorial_directory.js#L236), in this way the dataSourceSelection would be missing in the component.  

So I'm thinking that in order to safely share  dataSourceSelection instance among all components and be compatible with all usage ways, what about using a getter in each component to get instance directly (such as https://github.com/opensearch-project/OpenSearch -Dashboards/pull/6827/files#diff-ec5b6ef64504177e50cad326dcc07ada2135faafe924d25b0f4df088bc7b4823R80), so that I no longer have to manually pass dataSourceSelection through `ui.DataSourceSelector` and `ui.getDataSourceMenu`.

Reply:
Using getter sounds good